dagmc/dagnav.py

```diff
@@ -90,6 +90,25 @@ def geom_dimension_tag(self):
             create_if_missing=True,
         )
 
+    def add_volume_groups(self, group_map):
+        """Adds groups of volumes to the model.
+
+        Parameters
+        ----------
+        group_map : dict
+            A dictionary mapping group names to a 2-tuple of (int, list) whose
+            first element is the group ID and the second element is a list of
+            volume IDs or Volume objects in the group.
+        """
+        for group_name, (group_id, volumes) in group_map.items():
+            group = Group.create(self, name=group_name, group_id=group_id)
+
+            for volume in volumes:
+                if isinstance(volume, Volume):
+                    group.add_set(volume)
+                else:
+                    group.add_set(self.volumes[volume])
```

Comment:
This allows mix-and-match of volume IDs and `Volume` objects in a single list/dict... that could be confusing, but may it's OK?

Reply:
Sounds good - I'll add a test. I wasn't sure what the more popular use-case would be (I would probably just use IDs). This would allow for both without having to explicitly validate all the entries in the dictionary. I'm happy to add more input validation or change the data that is getting passed based on people's feedback (For instance, I'm not sure if the group ID is a good idea or bad idea. i.e. would we rather autogenerate group IDs similar to the `IDManager` in OpenMC?)

Reply:
I'm good with a mix of `Volume` objects and IDs. I think it's rare that a mix would be used, but if both are present the volume IDs still get validated by nature of a lookup on the `DAGModel.volumes` property.

I think auto-generation of Group IDs is a good idea. Maybe we can get to that soon.